File: frontend-app/profile-page/src/model/AuthenticatedUserDTO.ts
export class AuthenticatedUserDTO {
    authenticated: boolean;
    id: number;
    name: string;
    surname: string;
    email: string;
    phone: string;
    birthDate: string;
    userRoles: string[];

    constructor(
        authenticated: boolean,
        id: number,
        name: string,
        surname: string,
        email: string,
        phone: string,
        birthDate: string,
        userRoles: string[]
    ) {
        this.authenticated = authenticated;
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.phone = phone;
        this.birthDate = birthDate;
        this.userRoles = userRoles;
    }
}

File: api/shared/auth-commons/src/main/java/com/carrental/commons/authentication/model/AuthenticatedUserDTO.java
package com.carrental.commons.authentication.model;

import lombok.Data;

import java.util.List;

@Data
public class AuthenticatedUserDTO {

    private boolean isAuthenticated;

    private Long id;

    private String name;

    private String surname;

    private String email;

    private String phone;

    private String birthDate;

    private List<String> userRoles;
}
